src/panels/lovelace/editor/card-editor/hui-dialog-edit-card.ts

```diff
@@ -112,6 +112,7 @@ export class HuiDialogEditCard extends LitElement {
                       .lovelace="${this._params.lovelaceConfig}"
                       .value="${this._cardConfig}"
                       @config-changed="${this._handleConfigChanged}"
+                      @GUImode-changed=${() => this.requestUpdate()}
```

Comment:
Don't inline this, and why do we call `requestUpdate`? The event contains the current mode, we can use that to update the element?